### ISSUE TITLE
scapy.layers.inet import error on macOS 10.15.7

### ISSUE THREAD

**Issue by dantebben** [secdev/scapy#3246]
#### Brief description

Cannot import from `scapy.layers.inet` on macOS Catalina 10.15.7.  Get an Import Error:  Cannot import AnsweringMachine

This affects importing  `IP`, `Ether`, `UDP`, `TCP`, etc. from `scapy.layers.inet`.  A similar error message is produced when trying to import `PPP`, `HDLC`, etc. from `scapy.layers.ppp`.

#### Environment

-   Scapy version: 2.4.5 (also tried with 2.4.0, 2.4.1)
-   Python version: 3.9.5 (also tried with 3.9.1 and 3.8.10)
-   Operating System: macOS 10.15.7 Catalina

I have tried this on two MacBookPros, a fresh VM, plus a brand new Mac Mini that I just received.  Everything is running macOS 10.15.7.  This was working perhaps a month ago, so I tried rolling back to a previous version of Scapy (2.4.0, 2.4.1) and also previous versions of Python (3.9.1).  In each case the results are the same.  I do not see this issue on Ubuntu 20.04.  

#### How to reproduce

```python
from scapy.layers.inet import IP
```

#### Actual result

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/xxxxxxx/test_scapy/venv/lib/python3.9/site-packages/scapy/layers/inet.py", line 22, in <module>
    from scapy.ansmachine import AnsweringMachine
  File "/Users/xxxxxx/test_scapy/venv/lib/python3.9/site-packages/scapy/ansmachine.py", line 20, in <module>
    from scapy.sendrecv import send, sniff
  File "/Users/xxxxxx/test_scapy/venv/lib/python3.9/site-packages/scapy/sendrecv.py", line 61, in <module>
    import scapy.route  # noqa: F401
  File "/Users/xxxxxx/test_scapy/venv/lib/python3.9/site-packages/scapy/route.py", line 218, in <module>
    conf.route = Route()
  File "/Users/xxxxxx/test_scapy/venv/lib/python3.9/site-packages/scapy/route.py", line 37, in __init__
    self.resync()
  File "/Users/xxxxxx/test_scapy/venv/lib/python3.9/site-packages/scapy/route.py", line 45, in resync
    from scapy.arch import read_routes
  File "/Users/xxxxxx/test_scapy/venv/lib/python3.9/site-packages/scapy/arch/__init__.py", line 130, in <module>
    from scapy.arch.bpf.supersocket import *  # noqa F403
  File "/Users/xxxxxx/test_scapy/venv/lib/python3.9/site-packages/scapy/arch/bpf/supersocket.py", line 27, in <module>
    from scapy.layers.l2 import Loopback
  File "/Users/xxxxxx/test_scapy/venv/lib/python3.9/site-packages/scapy/layers/l2.py", line 17, in <module>
    from scapy.ansmachine import AnsweringMachine
ImportError: cannot import name 'AnsweringMachine' from partially initialized module 'scapy.ansmachine' (most likely due to a circular import) (/Users/xxxxxx/test_scapy/venv/lib/python3.9/site-packages/scapy/ansmachine.py)
```

#### Expected result

No Error messages, just the IP class has been imported.

#### Related resources

I have found a few work arounds, but each requires changing a large set of Scapy files I have created over time.

##### Workaround 1
```python
from scapy.all import IP.  # this works, but an IDE such as PyCharm doesn't like it because scapy.all doesn't list IP
```

##### Workaround 2
```python
from scapy.arch import str2mac  # this works, but linters such as flake8 don't like it because of unused imports
from scapy.layers.inet import IP  
```

##### Workaround 3 (this one is strange)
When in an interactive environment such as iPython, using `tab-complete`, but not actually importing from scapy.arch
```python
from scapy.arch import st[TAB]  # then delete the line after the tab-complete options appear
from scapy.layers.inet import IP
```

**Comment by gpotter2** [comment]
Thanks for the report BTW. I'm looking into this

**Comment by dantebben** [comment]
For another datapoint, I have now also tested on macOS 10.16 Big Sur and the same errors and work-arounds exist.

**Comment by guedou** [comment]
Similar issue with Python2.7 on macOS 10.14:
```
```

**Comment by guedou** [comment]
This patch seems to fix the issue:
```
$ git diff
diff --git a/scapy/arch/bpf/supersocket.py b/scapy/arch/bpf/supersocket.py
index 8587e6fa..7bb8d18b 100644
--- a/scapy/arch/bpf/supersocket.py
+++ b/scapy/arch/bpf/supersocket.py
@@ -24,7 +24,6 @@ from scapy.error import Scapy_Exception, warning
 from scapy.interfaces import network_name
 from scapy.supersocket import SuperSocket
 from scapy.compat import raw
-from scapy.layers.l2 import Loopback
 
 
 if FREEBSD:
@@ -414,6 +413,7 @@ class L3bpfSocket(L2bpfSocket):
         # Apple are breaking these kexts in a future version of macOS... so
         # the problem will eventually go away. They already don't work on Macs
         # with Apple Silicon (M1).
+        from scapy.layers.l2 import Loopback
         if DARWIN and iff.startswith('tun') and self.guessed_cls == Loopback:
             frame = raw(pkt)
         else:
```

@dantebben can you give it a try?

**Comment by dantebben** [comment]
@guedou 

Sorry for not getting back sooner, but just now had a chance to try this.  The patch works for me on macOS Big Sur.

**Comment by guedou** [comment]
Thanks for your feedback.

**Comment by EthanZeigler** [comment]
It's been almost a year and the version of scapy on pypi still has this issue. @gpotter2 is there a reason it hasn't been fixed?

**Comment by gpotter2** [comment]
This is fixed on master and in 2.5.0.
